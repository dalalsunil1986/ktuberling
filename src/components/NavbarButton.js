import React from 'react';
import PropTypes from 'prop-types';


const NavbarButton = ({ disabled, onClick, text }) => (
  <li className={disabled ? 'disabled' : ''}>
    <a role="button" onClick={onClick}>{text}</a>
  </li>
);


NavbarButton.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
};


NavbarButton.defaultProps = {
  disabled: false,
  onClick: () => {},
};


export default NavbarButton;
