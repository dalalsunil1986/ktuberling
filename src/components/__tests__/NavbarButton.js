import React from 'react';
import renderer from 'react-test-renderer';

import NavbarButton from '../NavbarButton';


describe('NavbarButton', () => {
  it('render', () => {
    const tree = renderer.create(<NavbarButton text="text" />);

    expect(tree).toMatchSnapshot();
  });


  it('render disabled', () => {
    const tree = renderer.create(<NavbarButton disabled text="disabled" />);

    expect(tree).toMatchSnapshot();
  });


  it('onClick default', () => {
    const tree = renderer.create(
      <NavbarButton text="text" />,
    );

    tree.toJSON().children[0].props.onClick();

    expect(tree).toMatchSnapshot();
  });
});
