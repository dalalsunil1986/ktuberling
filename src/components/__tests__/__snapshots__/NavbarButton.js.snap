// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarButton onClick default 1`] = `
<li
  className=""
>
  <a
    onClick={[Function]}
    role="button"
  >
    text
  </a>
</li>
`;

exports[`NavbarButton render 1`] = `
<li
  className=""
>
  <a
    onClick={[Function]}
    role="button"
  >
    text
  </a>
</li>
`;

exports[`NavbarButton render disabled 1`] = `
<li
  className="disabled"
>
  <a
    onClick={[Function]}
    role="button"
  >
    disabled
  </a>
</li>
`;
